### ISSUE TITLE
itunes_transporter.rb doesn't handle failures during upload very well any more

### ISSUE THREAD

**Issue by janpio** [fastlane/fastlane#13305]
Seems since the rework of `itunes_transporter.rb` for Windows compat the error message or stack trace you get when something fails during upload is not optimal any more.

Examples:

https://github.com/fastlane/fastlane/issues/13284#issue-357034799
```
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `chdir'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `cd'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
```

https://github.com/fastlane/fastlane/issues/13281#issuecomment-418806266
```
[11:45:17]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[11:45:24]: Login to App Store Connect (********)
[11:45:37]: Login successful

+--------------------------------------+--------------------------------------+
|                           deliver 2.103.1 Summary                           |
+--------------------------------------+--------------------------------------+
| ipa                                  | **********                           |
| force                                | true                                 |
| skip_screenshots                     | true                                 |
| skip_metadata                        | true                                 |
| username                             | **********                           |
| app_identifier                       | **********                           |
|                                      | iphone                               |
| run_precheck_before_submit           | false                                |
| screenshots_path                     | ./screenshots                        |
| metadata_path                        | ./metadata                           |
| app_version                          | 999999                               |
| platform                             | ios                                  |
| edit_live                            | false                                |
| use_live_version                     | false                                |
| skip_binary_upload                   | false                                |
| skip_app_version_update              | false                                |
| overwrite_screenshots                | false                                |
| submit_for_review                    | false                                |
| reject_if_possible                   | false                                |
| automatic_release                    | false                                |
| phased_release                       | false                                |
| precheck_default_rule_level          | warn                                 |
| ignore_language_directory_validatio  | false                                |
| n                                    |                                      |
| precheck_include_in_app_purchases    | true                                 |
+--------------------------------------+--------------------------------------+

[11:45:45]: Making sure the latest version on App Store Connect matches '999999' from the ipa file...
[11:45:49]: '999999' is the latest version on App Store Connect
[11:45:57]: Uploading binary to App Store Connect
[11:46:49]: Going to upload updated app to App Store Connect
[11:46:49]: This might take a few minutes. Please don't interrupt the script.
[11:46:58]: [Transporter Error Output]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
[11:46:58]: Transporter transfer failed.
[11:46:58]:
[11:46:58]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)

Looking for related GitHub issues on fastlane/fastlane...

➡️  AGV enabled check fails
    https://github.com/fastlane/fastlane/issues/9506 [closed] 33 💬
    04 Oct 2017

➡️  The App Name you entered has already been used.
    https://github.com/fastlane/fastlane/issues/8079 [closed] 6 💬
    03 May 2017

➡️  pilot crashes on upload
    https://github.com/fastlane/fastlane/issues/4821 [closed] 24 💬
    28 Jan 2017

and 3 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60zero%3F%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/2.2.0/fileutils.rb:127:in `chdir'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/2.2.0/fileutils.rb:127:in `cd'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/deliver/lib/deliver/runner.rb:160:in `upload_binary'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/deliver/lib/deliver/runner.rb:42:in `run'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/deliver/lib/deliver/commands_generator.rb:79:in `block (2 levels) in run'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/deliver/lib/deliver/commands_generator.rb:182:in `run'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/deliver/lib/deliver/commands_generator.rb:17:in `start'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:100:in `take_off'
	from /Users/jhoon/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /Users/jhoon/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/jhoon/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'
```

https://github.com/fastlane/fastlane/issues/13284#issuecomment-419014887
```
[15:32:54]: [Transporter Error Output]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
[15:32:54]: Transporter transfer failed.
[15:32:54]:
[15:32:54]: Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com (-22938)
...
/Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /Users/zolotyx/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:128:in `chdir'
	from /Users/zolotyx/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/fileutils.rb:128:in `cd'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:24:in `block (2 levels) in parsing_binding'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/bin/fastlane:23:in `load'
	from /Users/zolotyx/.rvm/gems/ruby-2.3.1/bin/fastlane:23:in `<main>'
```

https://github.com/fastlane/fastlane/issues/13284#issuecomment-419028920
``` 
[17:22:47]: fastlane detected a Gemfile in the current directory
[17:22:47]: however it seems like you don't use `bundle exec`
[17:22:47]: to launch fastlane faster, please use
[17:22:47]: 
[17:22:47]: $ bundle exec fastlane upload_testflight ipa:myapp.ipa account_type:account # this a custom lane called upload_testflight
[17:22:47]: 
[17:22:47]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[17:22:48]: ----------------------------------------
[17:22:48]: --- Step: Verifying fastlane version ---
[17:22:48]: ----------------------------------------
[17:22:48]: Your fastlane version 2.103.1 matches the minimum requirement of 1.106.1  ✅
[17:22:48]: ------------------------------
[17:22:48]: --- Step: default_platform ---
[17:22:48]: ------------------------------
[17:22:48]: Driving the lane 'ios upload_testflight' 🚀
[17:22:48]: -------------------
[17:22:48]: --- Step: pilot ---
[17:22:48]: -------------------
[17:22:49]: Login to App Store Connect (xxx@xxx.com)
[17:22:52]: Login successful
[17:22:53]: Ready to upload new build to TestFlight (App: xxxx)...
[17:22:55]: Going to upload updated app to App Store Connect
[17:22:55]: This might take a few minutes. Please don't interrupt the script.
[17:24:03]: [Transporter Error Output]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
[17:24:03]: Transporter transfer failed.
[17:24:03]: 
[17:24:03]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
+------------------+-----------------------+
|               Lane Context               |
+------------------+-----------------------+
| DEFAULT_PLATFORM | ios                   |
| PLATFORM_NAME    | ios                   |
| LANE_NAME        | ios upload_testflight |
+------------------+-----------------------+

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 💥   | pilot                      | 74          |
+------+----------------------------+-------------+

[17:24:03]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  [Regression] Build Fails with an Invalid 2 step response Error
    https://github.com/fastlane/fastlane/issues/13281 [closed] 31 💬
    10 hours ago

➡️  AGV enabled check fails
    https://github.com/fastlane/fastlane/issues/9506 [closed] 33 💬
    04 Oct 2017

➡️  The App Name you entered has already been used.
    https://github.com/fastlane/fastlane/issues/8079 [closed] 6 💬
    03 May 2017

and 4 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60zero%3F%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `chdir'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `cd'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:41:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by janpio** [comment]
This looks like a reproducible variant to cause this stacktrace:

```
[17:22:47]: $ bundle exec fastlane upload_testflight ipa:myapp.ipa account_type:account # this a custom lane called upload_testflight
...
[17:22:48]: Driving the lane 'ios upload_testflight' 🚀
[17:22:48]: -------------------
[17:22:48]: --- Step: pilot ---
[17:22:48]: -------------------
[17:22:49]: Login to App Store Connect (xxx@xxx.com)
[17:22:52]: Login successful
[17:22:53]: Ready to upload new build to TestFlight (App: xxxx)...
[17:22:55]: Going to upload updated app to App Store Connect
[17:22:55]: This might take a few minutes. Please don't interrupt the script.
[17:24:03]: [Transporter Error Output]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
[17:24:03]: Transporter transfer failed.
[17:24:03]: 
[17:24:03]: ERROR ITMS-90189: "Redundant Binary Upload. You've already uploaded a build with build number '1721' for version number '1.21.1'. Make sure you increment the build string before you upload your app to App Store Connect. Learn more in Xcode Help (http://help.apple.com/xcode/mac/current/#/devba7f53ad4)."
...
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:67:in `execute': [!] undefined method `zero?' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:287:in `block in execute'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `chdir'
	from /usr/local/Cellar/ruby/2.4.2_1/lib/ruby/2.4.0/fileutils.rb:120:in `cd'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:286:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/itunes_transporter.rb:380:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/pilot/lib/pilot/build_manager.rb:38:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:41:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:74:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.103.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```
Call `pilot` with a binary that is already uploaded to Testflight, which causes itunes_transporter.rb to get an error from iTMS that is then poorly handled.

**Comment by janpio** [comment]
Ah, forgot to mention why this causes problems: Because code execution stops with this error the nice error handling or manual input of password is not triggered any more, causing the regressions the users above reported.

**Comment by GregTheGreek** [comment]
This is failing for me on deliver. Getting a very similar output.

**Comment by joshdholtz** [comment]
Working on a fix for this!

**Comment by GregTheGreek** [comment]
I'm not sure if anyone has had this issue but lately, after using pilot it
waits for processing for up to an hour. Which I feel must be some bug
because I can manually upload much faster than that.

**Comment by janpio** [comment]
This should be fixed now in 2.104.0 🚀 Please update and let us know if this solved the problem for you.
